salt/metalk8s/orchestrate/downgrade/init.sls

```diff
@@ -10,6 +10,20 @@ Execute the downgrade prechecks:
         orchestrate:
           dest_version: {{ dest_version }}
 
+# In salt 2018.3 we can not do synchronous pillar refresh, so add a sleep
+# see https://github.com/saltstack/salt/issues/20590
+Wait for pillar refresh to complete:
+  salt.function:
+    - name: saltutil.refresh_pillar
+    - tgt: '*'
+    - require:
+      - salt: Execute the downgrade prechecks
+  module.run:
+    - test.sleep:
+      - length: 20
```

Comment:
20 seconds for a pillar refresh?! Isn't that a bit too much?

Reply:
Since we do several call to apiserver that may be on another machine I think it's quite ok and it's only one time during all the upgrade/downgrade process so it's ok IMO

Reply:
OK. We should do some profiling on our various Salt orchestrations, some time (not urgent at all, but would be interesting for us).